SU-SWS/react_paragraphs
# Keep all the users changes if the form does not validate

# READY FOR REVIEW

# Summary
- If the entity form doesn't validate, return the users data back to the UI.

# Need Review By (Date)
- 9/10

# Urgency
- high

# Steps to Test
1. checkout this branch
1. create some form validation in code using a `hook_form_alter` and throw a form error on anything something like:
```
function react_paragraphs_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
  $form['#validate'][] = 'break_it_all';
}
function break_it_all(&$form, \Drupal\Core\Form\FormStateInterface  $form_state){
  $form_state->setError($form, "Its broken");
}
```
1. Enter some data in the react items and save
1. validate you see your error and your data from the react items is still there.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
